### DISCUSSION TITLE
Feedback: Browser integration

### DISCUSSION THREAD

**Discussion by anugusaitejareddy** [mswjs/mswjs.io#387] [Q&A] [answered]
- [Open page](https://mswjs.io/docs/integrations/browser)

## Feedback

I want to know how to integrate msw with a react app?

**Comment by kettanaito** [comment, marked as answer]
Sharing my original response on Discord below.

Hi!
First things first,

> I don't know how I can run the server to mock my data.

You don't run servers with MSW. It either works in the browser alongside your app (https://mswjs.io/docs/integrations/browser), or in your server's process (https://mswjs.io/docs/integrations/node). Pick one or choose both at the same time. 

> I checked the examples but no one is using React

Here's an example that is using React:
https://github.com/mswjs/examples/tree/main/examples/with-remix

There's really nothing specific you should know in order to integrate MSW into a React project. The integration is almost the same in any framework, and the only thing that changes is where your framework exposes those entrypoints for the client and the server.
